controller.cpp

```diff
@@ -28,29 +27,47 @@ Ctrl* Ctrl::getInstance(){
   return instance;
 }
 
-bool Ctrl::runCalculation(){
+std::vector<std::tuple<std::string, int, double>> Ctrl::loadInputFiles(){
+  // create an instance of the model class
+  // ensures, that there is only ever one instance of the model class
+  if(current_file_loading == NULL){
+    current_file_loading = new Model();
+  }
+  std::vector<std::tuple<std::string, int, double>> atoms_for_list;
+  std::string atom_filepath = gui->getAtomFilepath();
+  std::string radius_filepath = gui->getRadiusFilepath();
+  // read atoms from file and save a vector containing the atoms
+  current_file_loading->readRadiiAndAtomNumFromFile(radius_filepath);
+  current_file_loading->listAtomTypesFromFile(atom_filepath);
+  for(auto elem : current_file_loading->number_of_atoms){
+    atoms_for_list.emplace_back(elem.first, elem.second, current_file_loading->radii[elem.first]);
+  }
+
+  return atoms_for_list;
+}
 
+bool Ctrl::runCalculation(){
   // create an instance of the model class
   // ensures, that there is only ever one instance of the model class
   if(current_calculation == NULL){
     current_calculation = new Model();
   }
 
   std::string atom_filepath = gui->getAtomFilepath();
-  std::string radius_filepath = gui->getRadiusFilepath();
-  // read atoms from file and save a vector containing the atoms
-  current_calculation->readRadiiAndAtomNumFromFile(radius_filepath);
+  std::unordered_map<std::string, double> radii_list;
+  gui->generateRadiiListFromGrid(radii_list);

```

Comment:
Why is this a `void` function? It seems like it should clearly have an `unordered_map` return value.

Reply:
Yes, I wanted to make an unordered_map function but the compiler gave me errors that I didn't understand. As a workaround, I made a void function with an unordered_map reference argument to modify the unordered_map directly.